packages/video_player/video_player/lib/video_player.dart

```diff
@@ -244,8 +261,30 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   @visibleForTesting
   int get textureId => _textureId;
 
+  /// Set the cache size in bytes. Default is maxSize of `100 * 1024 * 1024`
+  /// and maxFileSize of `10 * 1024 * 1024`.
+  ///
+  /// Cache used only for network data source.
+  ///
+  /// Throws StateError if you try to set the cache size twice. You can only set it once.
+  static void setCacheSize(int maxSize, int maxFileSize) {
+    assert(maxSize != null && maxSize > 0);
+    assert(maxFileSize != null && maxFileSize > 0);
+
+    if (_pluginInitializingCompleter != null) {
+      throw StateError(
+        "You can only set the VideoPlayerController cache size once.",
+      );
+    }
+
+    _maxCacheSize = maxSize;
+    _maxCacheFileSize = maxFileSize;
+  }
+
   /// Attempts to open the given [dataSource] and load metadata about the video.
   Future<void> initialize() async {
+    await _ensureVideoPluginInitialized();
```

Comment:
I'm hoping @cyanglaz or @ditman can say whether this is breaking. Before, we'd allow multiple calls to this method actually do work - if that was a bug this fixes that's fine, but if not this is breaking.